### DISCUSSION TITLE
为什么发送一些不合理的windows消息（不是关闭、销毁、退出窗口循环消息），PCL2、explorer、dev-cpp会崩，但是cmd、批处理程序、命令行程序不会呢？

### DISCUSSION THREAD

**Discussion by UserNameiiiii** [Meloong-Git/PCL#3043] [提问]
发送程序：
[窗口闪退.zip](https://github.com/Hex-Dragon/PCL2/files/13778506/default.zip)

**Comment by MoYuan-CN** [comment]
那些程序有非法信息处理或过滤吧？

**Reply by UserNameiiiii** [reply]
命令行程序（默认消息处理）还是我用C++自己写的呢，C++写的窗口程序（默认消息处理）又不行了，直接崩

**Reply by UserNameiiiii** [reply]
@JingHai-Lingyun  我觉得您的答案还是没有答疑解惑，为什么窗口程序会闪退，命令行程序则不会呢？

**Comment by WorldHim** [comment]
gui？

**Reply by UserNameiiiii** [reply]
cmd、批处理程序、命令行程序本质上也是图形用户界面啊，只是表示的方式不同而已

**Reply by WorldHim** [reply]
> cmd、批处理程序、命令行程序本质上也是图形用户界面啊，只是表示的方式不同而已

但是后三者在纯命令行环境可以运行

**Reply by UserNameiiiii** [reply]
不是啊，CMD是CMD，DOS是DOS

**Reply by WorldHim** [reply]
> 不是啊，CMD是CMD，DOS是DOS

纯命令行不代表 dos。。

**Reply by UserNameiiiii** [reply]
所以你说这个是干嘛，也不是答案...

**Comment by ghost** [comment]
因为cmd不会处理这类消息，就是他会抛弃这类信息继续运行

**Reply by allMagicNB** [reply]
不是抛弃，而是会返回报错。
